packages/@react-spectrum/s2/src/DateRangePicker.tsx

```diff
@@ -14,60 +14,286 @@ import {
+            ? state.granularity : undefined;
+        let showTimeField = !!timeGranularity;
+        return (
+          <>
+            <FieldLabel
+              isDisabled={isDisabled}
+              isRequired={isRequired}
+              size={size}
+              labelPosition={labelPosition}
+              labelAlign={labelAlign}
+              necessityIndicator={necessityIndicator}
+              contextualHelp={contextualHelp}>
+              {label}
+            </FieldLabel>
+            <FieldGroup
+              role="presentation"
+              shouldTurnOffFocusRing={buttonHasFocus}
+              isDisabled={isDisabled}
+              isInvalid={isInvalid}
+              size={size}
+              styles={style({
+                ...fieldInput(),
+                textWrap: 'nowrap',
+                paddingStart: 'edge-to-text',
+                paddingEnd: 4
+              })({size})}>
+              <div className={segmentContainer}>
+                <DateInput slot="start" className={segment}>
+                  {(segment) => <DateSegment className={dateInput} segment={segment} />}
+                </DateInput>
+                <span aria-hidden="true" className={style({flexShrink: 0, flexGrow: 0, paddingX: 2})}>–</span>
+                <DateInput slot="end" className={segment}>
+                  {(segment) => <DateSegment className={dateInput} segment={segment} />}
+                </DateInput>
+              </div>
+              {isInvalid && <div className={iconStyles}><FieldErrorIcon isDisabled={isDisabled} /></div>}
+              <div
+                className={style({
+                  flexShrink: 0,
+                  flexGrow: 1,
+                  display: 'flex',
+                  justifyContent: 'end'
+                })}>
+                <Button
+                  ref={buttonRef}
+                  // Prevent press scale from sticking while DateRangePicker is open.
+                  // @ts-ignore
+                  isPressed={false}
+                  onFocusChange={setButtonHasFocus}
+                  style={renderProps => pressScale(buttonRef)(renderProps)}
+                  className={renderProps => inputButton({
+                    ...renderProps,
+                    size,
+                    isOpen
+                  })}>
+                  <Provider
+                    values={[
+                      [IconContext, {
+                        styles: style({size: fontRelative(14)})
+                      }]
+                    ]}>
+                    <CalendarIcon />
+                  </Provider>
+                </Button>
+              </div>
+            </FieldGroup>
+            <Popover>
+              <div className={style({display: 'flex', flexDirection: 'column', gap: 16})}>
+                <RangeCalendar />
+                {showTimeField && (
+                  <div className={style({display: 'flex', gap: 16})}>
+                    <TimeField
```

Comment:
The time fields seem to be extra wide 
<img width="521" alt="image" src="https://github.com/user-attachments/assets/2d114824-3cbf-485e-b205-63e8fa152eb6" />